catalog/slack/post.js

```diff
@@ -14,22 +14,43 @@ function main(params) {
   var body = {
     channel: params.channel,
     username: params.username || 'Simple Message Bot',
-    text: params.text,
-    icon_emoji: params.icon_emoji
+    text: params.text
   };
 
+  if (params.icon_emoji) {
+    // guard against sending icon_emoji: undefined
+    body.icon_emoji = params.icon_emoji;
+  }
+
+  if (params.token) {
+    //
+    // this allows us to support /api/chat.postMessage
+    // e.g. users can pass params.url = https://slack.com/api/chat.postMessage
+    //                 and params.token = <their auth token>
+    //
+    body.token = params.token;
+  } else {
+    //
+    // the webhook api expects a nested payload
+    //
+    // notice that we need to stringify; this is due to limitations
+    // of the formData npm: it does not handle nested objects
+    //
+    console.log(body);
```

Comment:
for the webhook-style, it seems OK to leave it in. plus, as of now, the createSlackMessage test verifies the result by inspecting this log output.